pulp_rpm/app/tasks/synchronizing.py

```diff
@@ -331,6 +326,12 @@ def get_treeinfo_data(remote, remote_url):
         remote_url = fetch_remote_url(remote)
 
         if optimize and is_optimized_sync(repository, remote, remote_url):
+            pb_data = dict(
+                message="Optimized Sync (no changes)", code="sync.was_skipped", done=True
+            )
+            with ProgressReport(**pb_data) as pb:
+                pb.done = 1
+                pb.save()
```

Comment:
Is `save` needed inside the context manager and for a single iteration?
I understand that it was this way, I just checked and [the exit seems to do that](https://github.com/pulp/pulpcore/blob/354383883032277e7a1f7dc7ddf2dc0a5bc40fad/pulpcore/app/models/progress.py#L176).

Reply:
I think it's not necessary, yes.